crates/bevy_sprite/src/render/mod.rs

```diff
@@ -480,11 +480,15 @@ pub fn queue_sprites(
                 // By default, the size of the quad is the size of the texture
                 let mut quad_size = current_image_size;
 
-                // If a rect is specified, adjust UVs and the size of the quad
+                // If a rect is specified, adjust UVs and the size of the quad.
+                // The UVs are shrunk by half the size of a texel to prevent
+                // floating point inaccuracies from causing incorrect sampling.
                 if let Some(rect) = extracted_sprite.rect {
                     let rect_size = rect.size();
                     for uv in &mut uvs {
-                        *uv = (rect.min + *uv * rect_size) / current_image_size;
+                        *uv = (rect.min + *uv * rect_size
```

Comment:
IMO this computation is going to be much clearer if broken out across multiple lines.

Reply:
Made some changes but I'm not sure it came out nicer 😅. I'll come back to it later